### PR TITLE
Ci pin old serde v6.1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
None

Describe changes:
- make CI and oss-fuzz green again by pinning old serde

#11741 with requested change

### CODE REVIEW

**Review by victorjulien — APPROVED** [review]

File: rust/Cargo.toml.in [hunk 1]
```diff
@@ -65,6 +65,10 @@ ldap-parser = { version = "~0.4.0" }
 
 time = "~0.3.36"
 
+# pinned until oss-fuzz supports newer rust nightly cf https://github.com/google/oss-fuzz/pull/12365
+serde = { version = "=1.0.197" }
```
Comment (inashivb) [inline rust/Cargo.toml.in:69]: Is this `=` as a part of version string intentional? What does it mean?

**Comment by victorjulien** [comment]
Hmm is the CIFuzz build failure expected? It seems it still uses 209

**Comment by inashivb** [comment]
> Hmm is the CIFuzz build failure expected? It seems it still uses 209

yes from what I understand. It should get fixed once this is merged in master.
CIFuzz always runs master branch. (This is why Philippe is trying to get that clusterfuzzlite work in)

**Comment by catenacyber** [comment]
> Hmm is the CIFuzz build failure expected? It seems it still uses 209

Yes, CIFuzz uses what is in master, not in this PR

File: rust/Cargo.toml.in (hunk 1 above)
Comment (catenacyber) [reply rust/Cargo.toml.in:69]: yes, pin exact version https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#comparison-requirements

**Comment by victorjulien** [comment]
> > Hmm is the CIFuzz build failure expected? It seems it still uses 209
> 
> Yes, CIFuzz uses what is in master, not in this PR

I'm a bit confused now: if CIFuzz builds/uses master, what is the point of running it per PR then? Or am I missing something?

**Comment by catenacyber** [comment]
> > > Hmm is the CIFuzz build failure expected? It seems it still uses 209
> > 
> > 
> > Yes, CIFuzz uses what is in master, not in this PR
> 
> I'm a bit confused now: if CIFuzz builds/uses master, what is the point of running it per PR then? Or am I missing something?

CIFuzz is supposed to use the PR and not supposed to use master but it does now use master for our setup with master and main7 branch and 2 directories suricata and suricata7 and current python scripts in oss-fuzz cf `docker run --rm -t gcr.io/oss-fuzz/suricata python3 /opt/cifuzz/detect_repo.py --repo_name suricata` (we had 50% chance and got unlucky)

**Comment by victorjulien** [comment]
My FreeBSD runners fail this PR in `make distcheck`
image

**Comment by catenacyber** [comment]
Is this bug also in master and fixed by [11624](https://github.com/OISF/suricata/pull/11624) ? cc @jasonish

**Comment by victorjulien** [comment]
> Is this bug also in master and fixed by [11624](https://github.com/OISF/suricata/pull/11624) ? cc @jasonish

11624 doesn't even pass CI here, not going to bother running that. Master is fine.

**Comment by jasonish** [comment]
> Is this bug also in master and fixed by [11624](https://github.com/OISF/suricata/pull/11624) ? cc @jasonish

Doesn't look related.

**Comment by jasonish** [comment]
> My FreeBSD runners fail this PR in `make distcheck` image

This is likley due to the Cargo.lock and Cargo.toml not being in sync.  The change to the lock file is committed before the change to the toml file. The build system sees it as out of date?

**Comment by catenacyber** [comment]
> This is likley due to the Cargo.lock and Cargo.toml not being in sync.  The change to the lock file is committed before the change to the toml file. The build system sees it as out of date?

Trains something in https://github.com/OISF/suricata/pull/11763

**Comment by jasonish** [comment]
> > Is this bug also in master and fixed by [11624](https://github.com/OISF/suricata/pull/11624) ? cc @jasonish
> 
> 11624 doesn't even pass CI here, not going to bother running that. Master is fine.

Probably cause that commit forgot to update the lock file as well...  Should be fixed in https://github.com/OISF/suricata/pull/11762.

**Pull request closed without merging**